ids-webconsole/src/main/angular/src/app/keycerts/identityrenewest.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component } from '@angular/core';
+import { Title } from '@angular/platform-browser';
+import { ESTService } from './est-service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
+    templateUrl: './identityrenewest.component.html'
+})
+export class RenewIdentityESTComponent {
+    estUrl = 'https://daps-dev.aisec.fraunhofer.de';
+    rootCertHash = '7d3f260abb4b0bfa339c159398c0ab480a251faa385639218198adcad9a3c17d';
+    error = null;
+
+    constructor(private readonly titleService: Title,
+                private readonly estService: ESTService,
+                private readonly router: Router,
+                private readonly route: ActivatedRoute) {
+        this.titleService.setTitle('Renew Identity via the EST');
+    }
+
+    handleError(err: HttpErrorResponse) {
+        if (err.status === 0) {
+            this.error = 'Network Error';
+        } else {
+            const errObj = JSON.parse(err.error);
+            if (errObj.message) {
+                this.error = errObj.message;
+            } else {
+                // Errors have no message if it is disabled by the spring application
+                this.error = `Error response from connector: ${err.status}: ${errObj.error}`;
```

Comment:
Ist es sicher, dass es ein Feld `error` in dem Objekt gibt, wenn `message` nicht existiert? Was ist da dann drin und in welchem Format, hast du das getestet?

Reply:
Bei meinen Tests würde ein Json-Objekt bei Fehlern zurückgegeben, welches die HTTP-Response mit den Feldern `status` und `error` beschrieben hat (z.B. `{ "status": 400, "error": "Bad Request", ... }`). Die `message`-Property war dann die Beschreibung der Exception, wenn die entsprechende Application Property gesetzt ist. 

Angeblich sind die Nachrichten, die aus spring [ResponseStatusException](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/server/ResponseStatusException.html)s generiert werden RFC9457-kompatibel, aber das stimmt zumindest bei meinen Tests nicht. So ungefähr kann man aber die zurückgegebenen Json-Felder schon den Standardfeldern zuordnen.